sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -26,9 +27,21 @@ private static IEnumerable<Data> AsAmqpData(this IEnumerable<ReadOnlyMemory<byte
         {
             foreach (ReadOnlyMemory<byte> data in binaryData)
             {
+                ArraySegment<byte> segment;
+                if (!data.IsEmpty)
+                {
+                    if (!MemoryMarshal.TryGetArray(data, out segment))
+                    {
+                        segment = new ArraySegment<byte>(data.ToArray());
```

Comment:
I'm wondering if we should throw in this case;  it should never happen.  It would require there to be some data in the `ReadOnlyMemory<byte>` instance but that data is  not convertible to `byte`  _([ref](https://source.dot.net/#System.Private.CoreLib/MemoryMarshal.cs,308))_.

If that happens, I think it's a sign that something is wrong enough that we should fail.   Thoughts?